packages/components/pie-button/src/index.ts

```diff
@@ -244,9 +244,9 @@ export class PieButton extends LitElement implements ButtonProps {
             <button
                 @click=${this._handleClick}
                 class="o-btn"
-                type=${type}
-                variant=${variant}
-                size=${size}
+                type=${ifDefined(type)}
```

Comment:
Is there a way we can tell typescript that there will definitely be a value at this point? If there is no value by the time it gets to the template then something has gone wrong